core/math/vector2.h

```diff
@@ -150,7 +150,19 @@ _FORCE_INLINE_ Vector2 Vector2::plane_project(real_t p_d, const Vector2 &p_vec)
 	return p_vec - *this * (dot(p_vec) - p_d);
 }
 
-_FORCE_INLINE_ Vector2 operator*(real_t p_scalar, const Vector2 &p_vec) {
+_FORCE_INLINE_ Vector2 operator*(float p_scalar, const Vector2 &p_vec) {
+	return p_vec * p_scalar;
+}
+
+_FORCE_INLINE_ Vector2 operator*(double p_scalar, const Vector2 &p_vec) {
+	return p_vec * p_scalar;
+}
```

Comment:
Why does this need to be split in `float` and `double` variants?

Reply:
I suppose this is done so the multiplication can be done both using a float and double scalar, without incurring a performance loss due to automatic type casting.

Reply:
This is to resolve ambiguity for the compiler.

Reply:
But it's not needed for `Color` and `Quat`?

Reply:
No. The problem here is that Vector2 and Vector2i are interchangeable. So the compiler don't know if it should use the operator for Vector2 or the one for Vector2i. I'm not sure why it's not a problem with Vector3 though.